gobblin-audit/src/main/java/gobblin/audit/values/policy/column/DefaultColumnProjectionPolicyFactory.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+package gobblin.audit.values.policy.column;
+
+import java.lang.reflect.InvocationTargetException;
+
+import lombok.extern.slf4j.Slf4j;
+
+import org.apache.commons.lang3.reflect.ConstructorUtils;
+
+import com.google.common.base.Preconditions;
+import com.typesafe.config.Config;
+
+import gobblin.audit.values.auditor.ValueAuditGenerator;
+import gobblin.audit.values.auditor.ValueAuditRuntimeMetadata;
+import gobblin.util.ClassAliasResolver;
+
+/**
+ * Default factory class to create new {@link ColumnProjectionPolicy}s
+ */
+@Slf4j
+public class DefaultColumnProjectionPolicyFactory {
+
+  private static final String COLUMN_PROJECTION_POLICY_CLASS_NAME_KEY = ValueAuditGenerator.VALUE_AUDIT_CONFIG_KEY_PREFIX + "column.projection.class.name";
+
+  /**
+   * Create a new {@link ColumnProjectionPolicy} using the alias or cannonical classname specified at {@value #COLUMN_PROJECTION_POLICY_CLASS_NAME_KEY} in the <code>config</code>
+   * The {@link ColumnProjectionPolicy} class MUST have an accessible constructor <code>abc(Config config, TableMetadata tableMetadata)</code>
+   * <b>Note : Must have the key {@value #COLUMN_PROJECTION_POLICY_CLASS_NAME_KEY} set in <code>config</code> to create the {@link ColumnProjectionPolicy}</b>
+   * @see {@link AliasUtils}
+   *
+   * @param config job configs, Must have the key {@value #COLUMN_PROJECTION_POLICY_CLASS_NAME_KEY} set to create the {@link ColumnProjectionPolicy}
+   * @param tableMetadata runtime table metadata
+   *
+   * @return a new instance of {@link ColumnProjectionPolicy}
+   */
+  public static ColumnProjectionPolicy create(Config config, ValueAuditRuntimeMetadata.TableMetadata tableMetadata) {
+
+    Preconditions.checkArgument(config.hasPath(COLUMN_PROJECTION_POLICY_CLASS_NAME_KEY));
```

Comment:
Let's not use absolute paths for the Config properties, i.e the name should be something like "column.projection.class.name". Further, let's not abuse dots as a replacement of spaces. Dots should denote a logical scope. I think a better name is "columnProjection.class" or something along those lines.

Typesafe config allows for scoping. Let's assume the config object has already a restricted scope set by the caller of the constructor. This allows us to use the factory in different contexts which may have different fully-qualified property name.